presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -447,6 +454,7 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         Optional<Symbol> groupIdSymbol = Optional.empty();
         if (groupingSets.size() > 1) {
             groupIdSymbol = Optional.of(symbolAllocator.newSymbol("groupId", BIGINT));
+            symbolAllocator.addGroupIdSymbolForQuerySpecification(node, groupIdSymbol.get());
```

Comment:
Since `aggregate` is only use in `com.facebook.presto.sql.planner.QueryPlanner#plan(com.facebook.presto.sql.tree.QuerySpecification)`, then I think you can either:
* Pass `groupIdSymbol` to `aggregate(..)` as argument. If it's unused that shouldn't be a problem, but you can use such symbol later in `handleGroupings` method.
* Return some kind of `AggregateStatus` with both `planBuilder` and optional `groupIdSymbol`.

I think the I prefer the first approach. It's better than populating unrelated structures.